frontend/src/GameReportForm.tsx

```diff
@@ -24,10 +25,24 @@ import VictoryPoints from "./VictoryPoints";
 function GameReportForm() {
     const [
         formData,
-        { errorOnSubmit },
+        { errorOnSubmit, isFateOfEreborSelected, isCitiesSelected },
         { handleInputChange, validateField, handleSubmit },
     ] = useFormData();
 
+    const freeStrongholdOptions: Stronghold[] = staticFreeStrongholds.slice();
+    const shadowStrongholdOptions: Stronghold[] = shadowStrongholds.slice();
+
+    if (isFateOfEreborSelected) {
```

Comment:
The hard-coding here seems a little like a code-smell, especially since this data is also kinda contained in the mapping of `conditionalCities` later.

Do you think it would be cleaner to have some kind of functional approach here? The valid capturable territory is really just a function of what expansions are being played, after all... (and only two expansions affect it: Cities and Fate of Erebor)

Reply:
Good call! I took a stab at this and I feel alright about how `freeStrongholdOptions` and `shadowStrongholdOptions` are derived now, but wasn't sure if there was a better way to handle the`isFateOfEreborSelected` and `isCitiesSelected` variables still hanging around in `useFormData` (for the purpose of passing in `useEffect` conditions). How do you feel about it, was it a code smell there too or just in combination with the messy side effect stuff in `GameReportForm`?